STYLEGUIDE.md

```diff
@@ -0,0 +1,221 @@
+* [interfacename](_csharp-api/packagename/interfaces/interfacename/)
+* ...
+## Exception Classes
+* [exceptionname](_csharp-api/packagename/exceptions/exceptionname/)
+* ...
+```
+>Each of the sections (at both levels) should have a valid link to their respective info pages.
+
+The package information header should contain the following:
+```md
+Current Version (Stable branch): v0.0.0.0
+<Brief description of the package and its contents>
+```
+___
+
+## Classes (Base, Child, Templated, and Exceptions)
+
+Unlike the root and package level READMEs, the class README will contain more than just links to child folders. Each class will start with a **Class Overview Panel**:
+
+### [Parent Object Name](EXAMPLE)
+# Class Name
+
+This is a description of the class and what it does.
+
+<div style= "font-size:12px">
+
+| Using / Namespaces | <div style='color:yellow'>Taleworlds.Core</div> | <div style='color:green'>Saveable</div> |
+|:---|---|---:|
+| [System](EXAMPLE) | | 10000 |
+| [System.Xml](EXAMPLE)| | |
+| [TaleWorlds.SaveSystem](EXAMPLE) | | |
+</div>
+
+> For the non-savable classes, use ```'color:red'``` and change to text to "Not Savable".
+
+After this class overview, a small code snippet should exist showing how to use / initialize the class. **This should not be exhaustive**, just a quick reference.
+```csharp
+new InformationMessage("Hello World!")
+```
+
+The above would be a valid snippet for the ```InformationMessage``` Class, but can be longer if the extra length is used to give important context about the snippet.
+
+Directly after this intro snippet, we have a section outlining the variables of the class. All variables are assumed to have private setters unless otherwise noted. Each variable should contain all of the following information:
+
+* Name
+* Type (with link to that object or objects)
+* Notes
+
+## Variables
+
+<div style= "font-size:12px">
+
+| Mod | Name | Type | Notes |
+| --- | --- | --- | --- |
+| Const | HeadArmor | int |
+| Instance |MeshesMask | [SkinMask](EXAMPLE) | |
+| ReadOnly |BeardCoverType | [ArmorComponent](EXAMPLE).[BeardCoverTypes](EXAMPLE)
+| Static |ReinsRopeMesh | string | returns ```this.ReinsMesh + "_rope";```|
+|...|...|...|...|
+</div>
+
+Methods should follow parameters, seperated by a H1 header.
+
+## Methods
+
+<div style= "font-size:12px">
+
+| Type | Method Name | Number of Params |
+| --- | --- | --- |
+| void | [SetItem](EXAMPLE) | 2 |
+| [Equipment](EXAMPLE) | [Clone](EXAMPLE) | 1 | 
+| void | [FillFrom](EXAMPLE) | 2 |
+| ... | ... | ... |
+</div>
+
+
+> The class page should only list out the methods like a table of contents, the detail should be saved for the methods own README page.
+
+Methods should be like a miniature version of a class page, with a small table at the top for parameters (if any), return type, description, etc. as shown below:
+
+<div style= "font-size:12px">
+
+### [Parent Object Name](EXAMPLE)
+## Method Name
+
+| Parameters | Type | default | Optional? |
```

Comment:
MnB2 use C#, not TS. Type -> Parameters is better :)
Methods discription place in separate file, do I understand correctly?
In that case we don't need table here, because we have almost the same in another place. In addition, discription would be better placed here, not in another header.
Example: https://docs.python.org/3/library/math.html

Reply:
Also, Optional and default have the same meaning.

Reply:
oof. Not a C# master by any means, Fixed.

Reply:
Each file (Classes, Methods, etc) will focus on itself. So the Methods will have their description in their own file, not in their class file. I agree that this table could be optional, but users may navigate straight to a method from very far away (so to speak) in the documentation and won't arrive by going directly through their parrent class (or struct, w/e) tree. If you think this is still worth removing after that consideration, I'll remove.

Reply:
To summarize, I suggest:
Put methods declaration in class (when I read about class, I want to see it interface) as list (not table, because table assume that datas similar by size)
Put methods definition in separate file (without table).
In that case class won't be overwhelmed and definition won't be stretched out.